src/wallet/walletdb.cpp

```diff
@@ -1085,7 +1085,97 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
         }
     }
 
-    // TODO: Upgrade to using KeyMan
+    // Upgrade to using KeyMan
+    // Find the keys which are used in single key internal and external descriptors with
+    // the pre-taproot output types
+    if (pwallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS) &&
+        !pwallet->IsWalletFlagSet(WALLET_FLAG_USES_KEYMAN) &&
+        !pwallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)
+        && last_client <= 220000
+        ) {
+        std::map<CExtPubKey, std::pair<std::map<std::string, int>, uint64_t>> descs_keys;
+        std::map<std::string, int> tmpl = {{"pkh(", 0}, {"sh(wpkh(", 0}, {"wpkh(", 0}};
+
+        // Find xpubs used in pkh(), sh(wpkh()), and wpkh() descriptors
```

Comment:
1c76661: maybe clarify that we're dealing with the xpub at the root level, not the BIP44/49/84 "account" level.

Reply:
Done